compiler/rustc_target/src/abi/mod.rs

```diff
@@ -971,8 +971,14 @@ impl Niche {
         }
 
         // Compute the range of invalid values being reserved.
-        let start = v.end().wrapping_add(1) & max_value;
-        let end = v.end().wrapping_add(count) & max_value;
+        let (start, end, zero) = if *v.start() <= *v.end()
+            && *v.end() < (max_value - count)
+            && *v.start() != 0
+        {
+            (max_value - count, 0, true)
```

Comment:
Doesn't this reserve `count + 2` values instead of `count` values?

Reply:
I think you are right about that